### PR TITLE
Support global ordinals in `top_metrics`

### LINKED ISSUE #64774: Keyword support or alternative for top_metrics aggregation
In APM, we use a top_hits aggregation in some places to get certain values for the UI. E.g., if we are aggregation over `service.name`, we also need `agent.name` or each value of `service.name`. A `top_hits` aggregation can be costly, especially when there are many buckets.

Recently `top_metrics` was introduced as a lighter, faster alternative to top_hits. However, it does not support keywords (yet).

We would benefit from having top_metrics support keywords or a different aggregation that does the same, but for keywords.

### PR DESCRIPTION
**Pull request by nik9000**
Adds support for fetching `keyword` and `ip` fields in `top_metrics`.
This only works if we can get global ordinals for those fields so we
can't support the entire `keyword` family, but this gets us *somewhere*.

Closes #64774

### CODE REVIEW

File: x-pack/plugin/analytics/src/main/java/org/elasticsearch/xpack/analytics/topmetrics/TopMetricsAggregator.java [hunk 1]
```diff
@@ -241,44 +215,72 @@ public void close() {
...
+    
+    static MetricValues buildMetricValues(
+        ValuesSourceRegistry registry,
+        BigArrays bigArrays,
+        int size,
+        String name,
+        ValuesSourceConfig config
+    ) {
+        if (false == config.hasValues()) {
+            // `config` doesn't have the name if the
```
Comment (not-napoleon) [inline x-pack/plugin/analytics/src/main/java/org/elasticsearch/xpack/analytics/topmetrics/TopMetricsAggregator.java:259]:
```suggestion
            // `config` doesn't have the name if the field is unmapped
```

**Comment by nik9000** [comment]
run elasticsearch-ci/2

**Review by not-napoleon — APPROVED** [review]
I think this is fine.  I like that we're able to leverage the VSRegistry for top metrics now.  I'm still not sure the MultiValuesSourceConfig logic is the right long term solution, but I don't have a better plan and I have no interest in digging into it now, so there's no reason to block on that.

**Comment by nik9000** [comment]
> I think this is fine. I like that we're able to leverage the VSRegistry for top metrics now. I'm still not sure the MultiValuesSourceConfig logic is the right long term solution, but I don't have a better plan and I have no interest in digging into it now, so there's no reason to block on that.

Yeah. I didn't have any good ideas other than "don't do this now and tangle it up in support for bytes". It can come later, yeah.

**Pull request merged**